.github/workflows/ci.yml

```diff
@@ -19,172 +19,318 @@ jobs:
-      - run: pip install -r requirements.txt
-      - run: bash run-tests.sh ${{matrix.engine}} -k "TestDataFrame${{matrix.part}}"
+      - shell: bash -l {0}
+        run: pytest modin/pandas/test/ -k "TestDataFrame${{matrix.part}}"
         if: matrix.part != 3
-      - run: python -m pytest modin/pandas/test/test_series.py
+      - shell: bash -l {0}
+        run: python -m pytest modin/pandas/test/test_series.py
         if: matrix.part == 3
-      - run: python -m pytest modin/pandas/test/test_rolling.py
+      - shell: bash -l {0}
+        run: python -m pytest modin/pandas/test/test_rolling.py
         if: matrix.part == 3
-      - run: python -m pytest modin/pandas/test/test_concat.py
+      - shell: bash -l {0}
+        run: python -m pytest modin/pandas/test/test_concat.py
         if: matrix.part == 3
-      - run: python -m pytest modin/pandas/test/test_groupby.py
+      - shell: bash -l {0}
+        run: python -m pytest modin/pandas/test/test_groupby.py
         if: matrix.part == 3
-      - run: python -m pytest modin/pandas/test/test_reshape.py
+      - shell: bash -l {0}
+        run: python -m pytest modin/pandas/test/test_reshape.py
         if: matrix.part == 3
-      - run: python -m pytest modin/pandas/test/test_general.py
+      - shell: bash -l {0}
+        run: python -m pytest modin/pandas/test/test_general.py
         if: matrix.part == 3
-      - run: python -m pytest modin/pandas/test/test_io.py
+      - shell: bash -l {0}
+        run: python -m pytest modin/pandas/test/test_io.py
         if: matrix.part == 3
-      - run: python -m pytest modin/experimental/pandas/test/test_io_exp.py
+      - shell: bash -l {0}
+        run: python -m pytest modin/experimental/pandas/test/test_io_exp.py
         if: matrix.part == 3
-      - run: bash <(curl -s https://codecov.io/bash)
+      - shell: bash -l {0}
+        run: bash <(curl -s https://codecov.io/bash)
+
   test-windows:
     needs: [lint-commit, lint-flake8, lint-black, test-api, test-headers]
     runs-on: windows-latest
     strategy:
       matrix:
-        python-version: ["3.6.x", "3.7.x"]
+        python-version: ["3.6", "3.7", "3.8"]
         engine: ["ray", "dask"]
         part: ["Reduction_A", "Reduction_B", "Binary", "MapMetadata", "UDF", "Default", "Window", "Indexing", "Iter", "JoinSort", 3]
     env:
       MODIN_ENGINE: ${{matrix.engine}}
       MODIN_MEMORY: 1000000000
     name: test-windows
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
         with:
           fetch-depth: 1
-      - uses: actions/setup-python@v1
+      - name: Cache conda
+        uses: actions/cache@v1
         with:
+          path: ~/conda_pkgs_dir
+          key: ${{ runner.os }}-python-${{ matrix.python-version }}-conda-${{ github.run_id }}-${{ hashFiles('environment.yml') }}
+      - name: Cache pip
+        uses: actions/cache@v1
+        with:
+          path: ~\AppData\Local\pip\Cache
```

Comment:
is `~\AppData` a real path for Windows?.. where did you get this, is it documented for GH cache?

Reply:
Yes, https://github.com/actions/cache/blob/main/examples.md#multiple-oss-in-a-workflow.